staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/status_test.go

```diff
@@ -85,3 +86,29 @@ func TestAPIStatus(t *testing.T) {
 		}
 	}
 }
+
+func TestUnexpectedErrorToAPIStatus(t *testing.T) {
+	err := errors.UnexpectedObjectError{}
+	actual := ErrorToAPIStatus(&err)
+	expected := &metav1.Status{
+		TypeMeta: metav1.TypeMeta{Kind: "Status", APIVersion: "v1"},
+		Status:   metav1.StatusFailure,
+		Code:     500,
+		Reason:   metav1.StatusReasonUnknown,
+		Message:  err.Error(),
+	}
+	if !reflect.DeepEqual(actual, expected) {
+		t.Errorf("%s: Expected %#v, Got %#v", actual, expected, actual)
+	}
+}
+
+func TestResourceVersionConflictsToAPIStatus(t *testing.T) {
+	err := storage.NewResourceVersionConflictsError("fake_key", 0)
+	actual := ErrorToAPIStatus(err)
+	expected := &errors.NewConflict(schema.GroupResource{}, "", err).ErrStatus
```

Comment:
When constructing the `expected` object I suggest to set it field by field for better visibility of actual field values (like reason and message).